org.eclipse.buildship.ui.test/src/main/groovy/org/eclipse/buildship/ui/view/execution/BaseExecutionViewTest.groovy

```diff
@@ -0,0 +1,114 @@
+
+import org.eclipse.debug.core.ILaunchConfiguration
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy
+import org.eclipse.debug.core.ILaunchManager
+import org.eclipse.swt.widgets.Tree
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTree
+import org.eclipse.ui.IWorkbenchPage
+import org.eclipse.ui.console.ConsolePlugin
+import org.eclipse.ui.console.IConsole
+import org.eclipse.ui.console.IConsoleConstants
+import org.eclipse.ui.console.IConsoleListener
+import org.eclipse.ui.console.IConsoleManager
+
+import org.eclipse.buildship.core.CorePlugin
+import org.eclipse.buildship.core.launch.GradleRunConfigurationAttributes
+import org.eclipse.buildship.core.util.gradle.GradleDistributionSerializer
+import org.eclipse.buildship.ui.console.GradleConsole
+import org.eclipse.buildship.ui.external.viewer.FilteredTree
+import org.eclipse.buildship.ui.test.fixtures.SwtBotSpecification
+import org.eclipse.buildship.ui.util.workbench.WorkbenchUtils
+
+abstract class BaseExecutionViewTest extends SwtBotSpecification {
+
+    ConsoleListener consoleListener
+    ExecutionsView view
+    SWTBotTree tree
+
+    def setup() {
+        runOnUiThread {
+            view = WorkbenchUtils.showView(ExecutionsView.ID, null, IWorkbenchPage.VIEW_ACTIVATE)
+
+            WorkbenchUtils.showView(IConsoleConstants.ID_CONSOLE_VIEW, null, IWorkbenchPage.VIEW_ACTIVATE)
+            ConsolePlugin.default.consoleManager.addConsoleListener(consoleListener = new ConsoleListener())
+        }
+    }
+
+    def cleanup() {
+        ConsolePlugin.default.consoleManager.removeConsoleListener(consoleListener)
+        removeCloseableGradleConsoles()
+    }
+
+    protected GradleConsole getActiveConsole() {
+        consoleListener.activeConsole
+    }
+
+    protected SWTBotTree getCurrentTree() {
+        Tree filteredTree = ((FilteredTree)view.currentPage.getPageControl()).viewer.tree
+        tree = new SWTBotTree(filteredTree)
+    }
+
+    protected void launchTask(String projectLoc, String task) {
+        def attributes = new GradleRunConfigurationAttributes(
+                [task],
+                projectLoc,
+                GradleDistributionSerializer.INSTANCE.serializeToString(GradleDistribution.fromBuild()),
+                "",
+                null,
+                [],
+                [],
+                true,
+                true,
+                false,
+                false,
+                false);
+
+        ILaunchConfiguration configuration = CorePlugin.gradleLaunchConfigurationManager().getOrCreateRunConfiguration(attributes)
+        ILaunchConfigurationWorkingCopy launchConfigurationWC = configuration.getWorkingCopy()
+        launchConfigurationWC.setAttribute( "org.eclipse.debug.ui.ATTR_LAUNCH_IN_BACKGROUND", true )
+        launchConfigurationWC.setAttribute( "org.eclipse.debug.ui.ATTR_CAPTURE_IN_CONSOLE", true )
+        launchConfigurationWC.setAttribute( "org.eclipse.debug.ui.ATTR_PRIVATE", true )
```

Comment:
Can you please explain why these attributes are necessary?

Reply:
unnecessary , I just copy from other place , sorry for that.